docker-compose.yml

```diff
@@ -9,16 +9,36 @@ services:
       POSTGRES_USER: wordle_user
     volumes: 
       - dbdata:/var/lib/postgresql/data
-  # server:
-  #   build:
-  #     context: wordle
-  #     dockerfile: Dockerfile
-  #   environment:
-  #     WHALEAPP__DB_HOST: db
-  #     WHALEAPP__DB_NAME: wordle_db
-  #     WHALEAPP__DB_USER: wordle_user
-  #     WHALEAPP__DB_PASSWORD: wordle_password
 
+    healthcheck:
+      test: ["CMD-SHELL", "pg_isready"]
+      interval: 10s
+      timeout: 5s
+      retries: 20
+
+  app:
+    build:
+      context: .
```

Comment:
Have a look at https://docs.docker.com/compose/compose-file/compose-file-v3/#context
The parameter `context` is a bit dodgy, but it roughly corresponds to the directory that contains the `Dockerfile`. Here, you use `.` which is the directory that contains `docker-compose.yml` and that doesn't contain any `Dockerfile`. So, when you instruct it to build the app you need to pass the path of the correct `Dockerfile` that can build the app, and the same for the server.

Reply:
Thank you Leo! I think I have fixed the paths so the docker-compose will run.  

For the DB I am getting the error:
`db_1      | 2023-01-01 16:15:26.698 UTC [48] FATAL:  role "root" does not exist`

For the app, it exits with a status of 1. 
Saying:
`app_1     | Error: Cannot find module '/usr/src/app/app.tsx'
app_1     |     at Module._resolveFilename (node:internal/modules/cjs/loader:995:15)
app_1     |     at Module._load (node:internal/modules/cjs/loader:841:27)
app_1     |     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
app_1     |     at node:internal/main/run_main_module:23:47 {
app_1     |   code: 'MODULE_NOT_FOUND',
app_1     |   requireStack: []`

I had a look at this, by trying to change the WORKDIR file path, however I'm not sure if this may need to point to where the exact code is, i.e. the "use-wordle.ts" file? 

The server seems to run correctly as it says `Example app listening on port 3002` but when I head to localhost:3002 with it up and running. I cannot access the page.